### PR TITLE
Adds a blog post on why TypeScript is using Gatsby

### PR DESCRIPTION
**Pull request by orta**
## Description

Adds a blog post after a recommendation by Dustin, which takes my more [informal TS infra update](https://github.com/microsoft/TypeScript-Website/issues/130#issuecomment-577445454) and makes it a bit more cohesive and useful globally.

I'm open to feedback on content of the post, I think I found a nice way to end it

I'm currently about 20m into loading `yarn start` on the website to verify it works, if you do PR deploys then maybe that might happen sooner, so I figured I'd just make the PR

### CODE REVIEW

File: docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+### Ecosystem Wins
+
+Gatsby has a massive community, and a lot of the time I'm able to just re-use someone elses work. The TypeScript team only has a few [unique](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/gatsby-remark-shiki/README.md) [needs](https://github.com/microsoft/TypeScript-Website/tree/v2/packages/typescript-vfs) in [our code samples](http://www.typescriptlang.org/v2/dev/twoslash/) but other than that, it's a pretty vanilla website.
+
+It's been great to just add a single line into the [`gatsby-config.js`](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/typescriptlang-org/gatsby-config.js), run `yarn add gatsby-plugin-something` and suddenly a whole new set of features are set up. Plugins tend to be small and easy to audit, I've used a few.
+
+### TypeScript Support
+
+While not "out-of-the-box", with a little elbow grease (as of early 2020) to get comprehensive TypeScript support:
```
Comment (ascorbic) [inline docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md]: We should ensure that this matches our current docs for adding TS support.

Comment (hashimwarren) [reply docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md]: @ascorbic - he's right. We don't have official docs for adding typescript support yet. Let's run this as-is

File: docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ createSite()
 
 Gatsby adds a GraphQL API which sits in-between the setup of the data and the generation of files in your static site. This abstraction provides a very strong separation of "setting up the site" vs "representation on the file system" which provides more places to introspect what is going on internally.
 
-What does this look like in practice? The server-side generation starts at [`gatsby-node.js`](https://github.com/microsoft/TypeScript-Website/blob/0afd526969d98c321787ab1962f72f9361ab54bd/packages/typescriptlang-org/gatsby-node.js) but an interesting example is how a TSConfig Reference page is set up:
+What does this look like in practice? The server-side generation starts at [`gatsby-node.`](https://github.com/microsoft/TypeScript-Website/blob/0afd526969d98c321787ab1962f72f9361ab54bd/packages/typescriptlang-org/gatsby-node.js) but an interesting example is how a TSConfig Reference page is set up:
```
Comment (orta) [inline docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md]: You might want to keep this one ^

File: docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+By not using any of the React `setState`-ish APIs, I can guarantee there is no "runtime" React rendering happening on a user's browser either. This means the HTML in the built file is exactly what someone will see whether they have JavaScript enabled or not. One advantage of this, has been that I can reliably run [`BackstopJS`](https://garris.github.io/BackstopJS/) to take screenshots of these static files to keep track of visual regressions as the site grows and others start to contribute.
+
+Would I recommend this constraint to people making Gatsby websites? Probably not - it's going against the grain (React is a really good tool) of how you're expected to use Gatsby. But the trade-off is worth it for me, and it lowers the barrier if a compiler engineer wants to contribute.
+
+### Speed
+
+I'm blown away by how fast Gatsby is for a user.
+
+The founder of Gatsby, Kyle Matthews [gave a great talk in 2017 on the ways in which Gatsby is fast](https://youtu.be/Gtd-Ht-D0sg?t=961) and [here more recently](https://www.youtube.com/watch?v=HQEotVfTXwk), in rough:
```
Comment (orta) [inline docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md]:
```suggestion
The founder of Gatsby, Kyle Mathews [gave a great talk in 2017 on the ways in which Gatsby is fast](https://youtu.be/Gtd-Ht-D0sg?t=961) and [here more recently](https://www.youtube.com/watch?v=HQEotVfTXwk), in rough:
```

His last name is [Mathews](https://twitter.com/kylemathews) - I got caught on this one

File: docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md (hunk 1 above)
Comment (ascorbic) [reply docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md]: @hashimwarren Let's fix that by making TypeScript work out of the box

Comment (ascorbic) [reply docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md]: But yes, not today!

**Comment by orta** [comment]
work, thanks folks

**Pull request merged**